### PR TITLE
http: allow 'complete' directive with status without adapted arguments

### PR DESCRIPTION
**Pull request by raboof**
References #3306

### CODE REVIEW

File: akka-http/src/main/scala/akka/http/scaladsl/server/directives/RouteDirectives.scala [hunk 1]
```diff
@@ -49,6 +49,14 @@ trait RouteDirectives {
   def complete(m: => ToResponseMarshallable): StandardRoute =
     StandardRoute(_.complete(m))
 
+  /**
+   * Completes the request using the given arguments.
+   *
+   * @group route
+   */
+  def complete[T](status: StatusCode, m: ToResponseMarshallable): StandardRoute =
```
Comment (raboof) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/RouteDirectives.scala:57]: This should probably be
```suggestion
  def complete[T](status: StatusCode, v: T)(implicit marshaller: ToEntityMarshaller[T]): StandardRoute =
```
and then we should put those together so we can call the regular `ctx.complete(ToResponseMarshallable)`

File: akka-http/src/main/scala/akka/http/scaladsl/server/directives/RouteDirectives.scala [hunk 2]
```diff
@@ -49,6 +49,22 @@ trait RouteDirectives {
   def complete(m: => ToResponseMarshallable): StandardRoute =
     StandardRoute(_.complete(m))
 
+  /**
+   * Completes the request using the given arguments.
+   *
+   * @group route
+   */
+  def complete[T](status: StatusCode, v: T)(implicit m: ToEntityMarshaller[T]): StandardRoute =
```
Comment (jrudolph) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/RouteDirectives.scala:57]: The main `complete` directive is by-name. Could the strict version lead to surprises?

File: akka-http/src/main/scala/akka/http/scaladsl/server/directives/RouteDirectives.scala [hunk 3]
```diff
@@ -49,6 +49,22 @@ trait RouteDirectives {
...
+   */
+  def complete[T](status: StatusCode, v: => T)(implicit m: ToEntityMarshaller[T]): StandardRoute =
+    StandardRoute(_.complete((status, v)))
+
+  /**
+   * Completes the request using the given arguments.
+   *
+   * @group route
+   */
+  def complete[T](status: StatusCode, headers: immutable.Seq[HttpHeader], v: T)(implicit m: ToEntityMarshaller[T]): StandardRoute =
```
Comment (jrudolph) [inline akka-http/src/main/scala/akka/http/scaladsl/server/directives/RouteDirectives.scala:65]:
```suggestion
  def complete[T](status: StatusCode, headers: immutable.Seq[HttpHeader], v: => T)(implicit m: ToEntityMarshaller[T]): StandardRoute =
```

Comment (jrudolph) [reply akka-http/src/main/scala/akka/http/scaladsl/server/directives/RouteDirectives.scala:65]: Incidentally, while I think it's good to have those as by-name for consistency, it's less likely that they are used in the wrong way (compared to the one-argument `complete`). The one argument `complete` could be used like that:


```scala
complete {
  // run some side-effects in here
}
```

but that's less likely with multiple arguments.

**Pull request merged**